tests/suites/test_suite_lms.function

```diff
@@ -149,18 +149,40 @@ exit:
 /* END_CASE */
 
 /* BEGIN_CASE */
-void lms_import_export_test (  data_t * pub_key )
+void lms_import_export_test (  data_t * pub_key, int expected_import_rc )
 {
     mbedtls_lms_public_t ctx;
-    uint8_t exported_pub_key[MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10)];
+    size_t exported_pub_key_buf_size = 0;
+    size_t exported_pub_key_size = 0;
+    unsigned char *exported_pub_key = NULL;
 
     mbedtls_lms_public_init(&ctx);
     TEST_EQUAL( mbedtls_lms_import_public_key( &ctx, pub_key->x, pub_key->len ), 0 );
-    TEST_EQUAL( mbedtls_lms_export_public_key( &ctx, exported_pub_key,
-                                                sizeof(exported_pub_key), NULL ), 0 );
 
-    ASSERT_COMPARE( pub_key->x, MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10),
-                    exported_pub_key, MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10) );
+    if( expected_import_rc == 0 )
+    {
+        exported_pub_key_buf_size = MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10);
```

Comment:
Since we expect the output from export to be the same as the input to import, the export buffer size should presumably be `pub_key->len`. And then we might assert after doing the export that `exported_pub_key_size == MBEDTLS_LMS_PUBLIC_KEY_LEN(MBEDTLS_LMS_SHA256_M32_H10)` because “exported keys must have a certain size” is a separate correctness property from “import-export must give the original string”.

Same in LMOTS.